### PR TITLE
[Color system] add critical action variants, simplify neutral actions, add surface actions

### PR DESCRIPTION
**Pull request by danrosenthal**

### CODE REVIEW

File: src/components/Badge/Badge.scss [hunk 1]
```diff
@@ -18,7 +18,7 @@ $pip-spacing: ($height - $pip-size) / 2;
   display: inline-flex;
   align-items: center;
   padding: 0 $horizontal-padding;
-  background-color: var(--p-interactive-neutral-elevation-1, color('sky'));
+  background-color: var(--p-neutral-action-disabled, color('sky'));
```
Comment (tmlayton) [inline src/components/Badge/Badge.scss]: feels like the wrong role to use here

File: src/components/Navigation/Navigation.scss [hunk 2]
```diff
@@ -166,7 +166,7 @@ $disabled-fade: 0.6;
       --p-none,
       linear-gradient(color(state, hover), color(state, hover))
     );
-    background-color: var(--p-interactive-neutral-elevation-2);
+    background-color: var(--p-neutral-action);
```
Comment (tmlayton) [inline src/components/Navigation/Navigation.scss]: I think these should be surface-based roles, more similar to action list.

File: src/components/Badge/Badge.scss (hunk 1 above)
Comment (danrosenthal) [reply src/components/Badge/Badge.scss]: I feel the same way, but I'm not sure what actually fits there.

File: src/components/Navigation/Navigation.scss (hunk 2 above)
Comment (danrosenthal) [reply src/components/Navigation/Navigation.scss]: Makes sense to me.

File: src/components/TopBar/components/SearchField/SearchField.scss [hunk 3]
```diff
@@ -99,7 +99,7 @@ $stacking-order: (
 
 .Clear {
   @include recolor-icon(
-    var(--p-interactive-neutral-elevation-4, color('ink', 'lightest'))
+    var(--p-neutral-action-pressed, color('ink', 'lightest'))
```
Comment (tmlayton) [inline src/components/TopBar/components/SearchField/SearchField.scss]: Also seems like the wrong role, maybe a new role for `neutral-icon`, `neutral-icon-hovered`, `neutral-icon-pressed`?

File: src/components/Badge/Badge.scss (hunk 1 above)
Comment (danrosenthal) [reply src/components/Badge/Badge.scss]: Probably a good use-case for the base neutral

```suggestion
  background-color: var(--p-neutral, color('sky'));
```

File: src/components/TopBar/components/SearchField/SearchField.scss (hunk 3 above)
Comment (danrosenthal) [reply src/components/TopBar/components/SearchField/SearchField.scss]: I don't think so, since this is in-effect an icon-only button

**Pull request merged**